crates/ibc/src/core/context.rs

```diff
@@ -0,0 +1,396 @@
+    ) -> Result<PacketCommitment, ChannelError>;
+
+    fn get_packet_receipt(
+        &self,
+        key: &(PortId, ChannelId, Sequence),
+    ) -> Result<Receipt, ChannelError>;
+
+    fn get_packet_acknowledgement(
+        &self,
+        key: &(PortId, ChannelId, Sequence),
+    ) -> Result<AcknowledgementCommitment, ChannelError>;
+
+    /// Compute the commitment for a packet.
+    /// Note that the absence of `timeout_height` is treated as
+    /// `{revision_number: 0, revision_height: 0}` to be consistent with ibc-go,
+    /// where this value is used to mean "no timeout height":
+    /// <https://github.com/cosmos/ibc-go/blob/04791984b3d6c83f704c4f058e6ca0038d155d91/modules/core/04-channel/keeper/packet.go#L206>
+    fn packet_commitment(
+        &self,
+        packet_data: Vec<u8>,
+        timeout_height: TimeoutHeight,
+        timeout_timestamp: Timestamp,
+    ) -> PacketCommitment {
+        let mut hash_input = timeout_timestamp.nanoseconds().to_be_bytes().to_vec();
+
+        let revision_number = timeout_height.commitment_revision_number().to_be_bytes();
+        hash_input.append(&mut revision_number.to_vec());
+
+        let revision_height = timeout_height.commitment_revision_height().to_be_bytes();
+        hash_input.append(&mut revision_height.to_vec());
+
+        let packet_data_hash = self.hash(packet_data);
+        hash_input.append(&mut packet_data_hash.to_vec());
+
+        self.hash(hash_input).into()
+    }
+
+    fn ack_commitment(&self, ack: Acknowledgement) -> AcknowledgementCommitment {
+        self.hash(ack.into()).into()
+    }
+
+    /// A hashing function for packet commitments
+    fn hash(&self, value: Vec<u8>) -> Vec<u8>;
+
+    /// Returns the time when the client state for the given [`ClientId`] was updated with a header for the given [`Height`]
+    fn client_update_time(
+        &self,
+        client_id: &ClientId,
+        height: Height,
+    ) -> Result<Timestamp, ChannelError>;
+
+    /// Returns the height when the client state for the given [`ClientId`] was updated with a header for the given [`Height`]
+    fn client_update_height(
+        &self,
+        client_id: &ClientId,
+        height: Height,
+    ) -> Result<Height, ChannelError>;
+
+    /// Returns a counter on the number of channel ids have been created thus far.
+    /// The value of this counter should increase only via method
+    /// `ChannelKeeper::increase_channel_counter`.
+    fn channel_counter(&self) -> Result<u64, ChannelError>;
+
+    /// Returns the maximum expected time per block
+    fn max_expected_time_per_block(&self) -> Duration;
+
+    /// Calculates the block delay period using the connection's delay period and the maximum
+    /// expected time per block.
+    fn block_delay(&self, delay_period_time: Duration) -> u64 {
+        calculate_block_delay(delay_period_time, self.max_expected_time_per_block())
+    }
+}
+
+pub trait ExecutionContext: ValidationContext {
```

Comment:
I think using `ValidationContext` as a supertrait here is convenient but might be confusing to implementors if it has methods such as `check_emitted_event()`, etc. which only make sense for VPs.

Reply:
As discussed [here](https://github.com/cosmos/ibc-rs/pull/241#discussion_r1026565196), I don't think we'll have Anoma VP-specific methods in `ValidationContext`. Still worth making sure that all methods in `ValidationContext` make sense in `ExecutionContext` though.